src/pages/data-first-and-data-last-a-comparison.md

```diff
@@ -0,0 +1,474 @@
+
+```reason
+let update = (~isAdmin=false, u: user) => { ... };
+```
+
+With data-first this is not possible. Because the data comes first and then all the optional values, we are forced to always include a `unit` type as last param to make sure the compiler knows when the function has been fully applied:
+
+```reason
+let update = (u: user, ~isAdmin=false, unit) => { ... };
+```
+
+### Less straight-forward composition
+
+As mentioned at the beginning, data-last allows to compose functions very ergonomically, as one can partially apply a function by passing some parameters to it, and leave the last one to be filled later on.
+
+With data-first, it is not that straight forward though. Adapting the original example with `plusOne` and `addOneToList`, we can see how partially applying the `map` call is not possible anymore, unless we use another function that flips the two parameters:
+
+```reason
+let plusOne = a => a + 1;
+let addOneToList = list => Belt.List.map(list, plusOne);
+                                           ↑      ↑
+                   /* Can't pass `plusOne` without passing `list` first */
+```
+
+To get around this problem, Reason introduced something called [pipe placeholders](https://reasonml.github.io/docs/en/pipe-first#pipe-placeholders) to indicate a positional argument that will be filled later:
+
+```reason
+let plusOne = a => a + 1;
+let addOneToList = Belt.List.map(_, plusOne);
+let listA = [1, 2, 3];
+let listB = addOneToList(listA); // [2, 3, 4]
+```
+
+### Usages in OCaml and other languages
+
+Data-first convention and pipe first operator are not maybe as traditional as data-last, but there are some other usages in functional and object-oriented languages.
+
+- [Elixir](https://hexdocs.pm/elixir/Kernel.html#%7C%3E/2) includes in the core language a pipe operator that passes the value as the first param called pipe-forward, and [its standard library](https://hexdocs.pm/elixir/List.html#summary) has been designed with data-first in mind.
+- There are some [commonly used libraries](https://ocaml.janestreet.com/ocaml-core/latest/doc/core_kernel/Core_kernel/Map/) in OCaml that have adopted a data-first approach in their design. Even some OCaml imperative APIs use it, like the ones found in [`Hashtbl`](http://caml.inria.fr/pub/docs/manual-ocaml/libref/Hashtbl.html).
+- In other cases, like C#, there are [proposals](https://github.com/dotnet/csharplang/issues/96) to include pipe-first operator in the future.
+
+## Conclusion
+
+So, if you have read until here, first, you're awesome 😄 and second, what could we conclude?
+
+From all the data seen above, there is no clear "better way", both data-first and data-last have their own set of trade-offs.
+
+Data-last has:
```

Comment:
Among these trade-offs it's important to clarify that data-last works specifically with `|>` and data-first works with `->`. And that `->`/`|.`/`_` (pipe placeholder) is not portable between other OCaml backends (or indeed to ReasonML targeting other backends). So if you'd like to keep more portability (say you want to keep open the possibility of doing fullstack ReasonML), there is a stronger case for data-last and `|>`.